.githooks/pre-commit

```diff
@@ -4,7 +4,14 @@
 echo INFO: Pre-commit hook running from $PWD
 git version
 
+# Stash any changes to the working tree that are not going to be committed
+git stash -q --keep-index
+
 #check that the EDAM source file is in the right format (RDF/XML)
 grep -c 'owl:Class' EDAM_dev.owl
 owlclassfound=$?
+
+# Unstash changes to the working tree that we had stashed
+git stash pop -q
```

Comment:
Note to myself to look into in the future:   (PLEASE DON'T RESOLVE)
This `git stash pop -q` sometimes returns 1-line outputs (`git version 2.29.2.windows.2`), like
-  `No stash entries found.`
-  `Already up to date!`

Reply:
I found something maybe this can help - 
`git stash pop`  throws away the (topmost, by default) stash after applying it, whereas `git stash apply ` leaves it in the stash list for possible later reuse (or you can then git stash drop it).

This happens unless there are conflicts after `git stash pop`, in which case it will not remove the stash, leaving it to behave exactly as `git stash apply`.

Another way to look at it: `git stash pop` is `git stash apply && git stash drop`.

So can we use - `git stash apply` in place of  `git stash pop -q`